build-farm/platform-specific-configurations/linux.sh

```diff
@@ -78,7 +78,18 @@ fi
 
 if [ "${ARCHITECTURE}" == "arm" ]
 then
-  export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --with-jobs=4 --with-memory-size=2000"
+  if lscpu | grep aarch64; then
+     echo Validating 32-bit environment on 64-bit host - perhaps it is a docker container ...
+     if ! file /bin/ls | grep "32-bit.*ARM"; then
+       echo /bin/ls is not a 32-bit system. This configuration is invalid without extra work
+       exit 1
+     fi
+     echo Looks reasonable - configuring to allow building that way ...
+     export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --target=armv7l-unknown-linux-gnueabihf --host=armv7l-unknown-linux-gnueabihf"
+  fi
+  if [ "$(lscpu|awk '/^CPU\(s\)/{print$2}')" = "4" ]; then
```

Comment:
4 seems a very specific number, what if there's 2 or 8 or whatever?

Reply:
Honeslty this was always a hack to make it use all the CPU cores on the restricted RAM scaleway systems that we have as we were ending up with the builds taking a lot longer because the build system was actively dropping the number of cores it would otherwise use. I'm propoing leaving that stuff as is to allow maximum use of the Scaleway systems when builds end up on there, and then probably revisit it later.

Bear in mind that the previous code ALWAYS locked the bulid system to 4 cores/2Gb on ARM so this change just allows it to use more if they are available. I'll be delighted to strip this out once we're running on better systems :-) If anyone wants to go back and verify the implications on the existing scaleway systems of removing this I'm happy for them to check it and potentially remove it ASAP but for now I think this is the right approach to ensure stability.

Reply:
If you add a comment explaining this situation then I'm good :-)

Reply:
Done - shortened the above essay to indicate that it is a temporary override for the Scaleway machine capacities.